packages/bibtex/init.lua

```diff
@@ -58,6 +68,56 @@ local parseBibtex = function (fn)
    return entries
 end
 
+--- Resolve the 'crossref' and 'xdata' fields on a bibliography entry.
+-- (Supplementing the entry with the attributes of the parent entry.)
+-- Once resolved recursively, the crossref and xdata fields are removed
+-- from the entry.
+-- So this is intended to be called at first use of the entry, and have no
+-- effect on subsequent uses: BibTeX does seem to mandate cross refererences
+-- to be defined before the entry that uses it, or even in the same bibliography
+-- file.
+-- Implementation note:
+-- We are not here to check the consistency of the BibTeX file, so there is
+-- no check that xdata refers only to @xdata entries
+-- Removing the crossref field implies we won't track its use and implicitely
+-- cite referenced entries in the bibliography over a certain threshold.
+-- @tparam table bib Bibliography
+-- @tparam string key Valid entry key
+local function crossrefAndXDataResolve (bib, key)
+   local entry = bib[key]
+   local refs
+
+   local xdata = entry.attributes.xdata
+   if xdata then
+      entry.attributes.xdata = nil
+      refs = xdata and pl.stringx.split(xdata, ",")
+   end
+
+   local crossref = entry.attributes.crossref
+   if crossref then
+      refs = refs or {}
+      table.insert(refs, crossref)
+      entry.attributes.crossref = nil
+   end
+
+   if not refs then
+      return
+   end
+   for _, ref in ipairs(refs) do
+      local parent = bib[ref]
+      if parent then
+         crossrefAndXDataResolve(bib, ref)
+         for k, v in pairs(parent.attributes) do
+            if not entry.attributes[k] and k ~= "crossref" and k ~= "xdata" then
+               entry.attributes[k] = v
```

Comment:
The Biblatex manual has an appendix B "Default Inheritance Setup" implying some fields are re-mapped depending on the target and source's types. This is also supposed to be fully configurable (with LaTeX commands, with "options" fields, etc.)
E.g. 
- the "title" of a `@periodiical̀` would become the "journaltitle" (an alias to "journal") of an `@article`
- the "title" of a `@book` would become the "booktitle" of an `@inbook`... but apparently not of an `@incollection`? Why?
 
This appendix table is really messy to read... Playing with a few JS-based bibtex engines, while they do supplement fields from `crossref` but do **not** seem to apply such field mappings... I sort of understand why!

The Biblatex manual is real hard to read... And it's never stated clearly if `xdata` follows the same inheritance mapping... Doh. This format is totally insane.

Reply:
Re: `crossref` mapping - Typst does some minimal/naive hard-coded inheritance:
https://github.com/typst/biblatex/blob/a43de9ca4fb5c981c39c4c162e6df54684015fc8/src/lib.rs#L630

Reply:
Minimal implementation done.